lib/birthday_reader.rb

```diff
@@ -1,12 +1,11 @@
+require 'yaml'
+
 class BirthdayReader
   def self.get_birthdays(file_path)
     birthdays = []
     if File.exist?(file_path) 
-      file = File.new(file_path, 'r')
-      file.each_line do |line|
-        birthdays << line.chomp.split
-      end
-      file.close
+	bday_list = YAML.load_file(file_path)
```

Comment:
I am not against the structuring.

What I am saying is that the purpose of the function is to read birthdays. It is not to read and filter. That is why I think we should not filter the birthdays here.

And because we are just managing strings of birthdays I guess memory is not a concern.